x/pool/keeper/msg_server.go

```diff
@@ -0,0 +1,92 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	"cosmossdk.io/x/pool/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type MsgServer struct {
+	Keeper
+}
+
+var _ types.MsgServer = MsgServer{}
+
+// NewMsgServerImpl returns an implementation of the distribution MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &MsgServer{Keeper: keeper}
+}
+
+func (k MsgServer) FundCommunityPool(ctx context.Context, msg *types.MsgFundCommunityPool) (*types.MsgFundCommunityPoolResponse, error) {
+	depositor, err := k.authKeeper.AddressCodec().StringToBytes(msg.Depositor)
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid depositor address: %s", err)
+	}
+
+	if err := validateAmount(msg.Amount); err != nil {
+		return nil, err
+	}
+
+	if err := k.Keeper.FundCommunityPool(ctx, msg.Amount, depositor); err != nil {
+		return nil, err
+	}
+
+	return &types.MsgFundCommunityPoolResponse{}, nil
+}
+
+func (k MsgServer) CommunityPoolSpend(ctx context.Context, msg *types.MsgCommunityPoolSpend) (*types.MsgCommunityPoolSpendResponse, error) {
+	if err := k.validateAuthority(msg.Authority); err != nil {
+		return nil, err
+	}
+
+	if err := validateAmount(msg.Amount); err != nil {
+		return nil, err
+	}
+
+	recipient, err := k.authKeeper.AddressCodec().StringToBytes(msg.Recipient)
+	if err != nil {
+		return nil, err
+	}
+
+	if k.bankKeeper.BlockedAddr(recipient) {
```

Comment:
https://github.com/cosmos/cosmos-sdk/blob/fdcb16111091a45c33c2d65e9ed8ca8b2ebb308f/x/bank/keeper/keeper.go#L267

this is checked already here no need to double check it